### ISSUE TITLE
Automatically use latest save

### ISSUE THREAD

**Issue by numbahs** [fribbels/Fribbels-Epic-7-Optimizer#44]
Automatically pull the latest export.json from saves folder and use that as the data so that we don't have to load the data every time. Or stash the latest changes that were made (outside of export.json too) and use that.

I think this was a thing before but now that there is the new export.json naming and new default saves folder, it isn't working anymore.

**Comment by fribbels** [comment]
Hi, it does automatically pull the latest autosave.json from your saves folder. Can you check if you have in one there, or if not just rename one of your saves to autosave.json.

**Comment by numbahs** [comment]
There is an autosave.json, but nothing is populated when my saves folder isn't the default saves folder.

**Comment by fribbels** [comment]
The default saves folder changed to Documents/FribbelsOptimizerSaves, where it was inside the app's install directory before. 

Could you open the developer tools, then hit CTRL + R to force reload the app, and show me any errors in the Console tab?

https://i.imgur.com/EOdY3As.png

https://i.imgur.com/gaeXA4S.png

**Comment by numbahs** [comment]
It is trying to take the autosave from the Documents/FribbledOptimizerSaves instead of the saves folder.

**Comment by fribbels** [comment]
Documents/FribbledOptimizerSaves **is** the saves folder now.

**Comment by numbahs** [comment]
I set the saves folder to my H drive via this

![image](https://user-images.githubusercontent.com/15134129/110225640-7e6a2d00-7eb5-11eb-85ec-396161fc12b0.png)

It is still trying to find the autosave in C:/Users/[User]/Documents/FribbelsOptimizerSaves/autosave.json, but that autosave is now in my H drive.

**Comment by fribbels** [comment]
Hmm interesting, Ill take a closer look when I get home, but could you show me the contents of your settings.ini file in either folder?

**Comment by numbahs** [comment]
Where would the settings.ini file be? Also I sent a screenshot already of the only error I was getting.

**Comment by fribbels** [comment]
It loads the settings.ini from your Documents/FribbelsOptimizerSaves folder. The settinngs.ini file tells the optimizer where the default save location is

![https://i.imgur.com/4v34B8S.png)(https://i.imgur.com/4v34B8S.png)

If you don't have one, try creating a settings.ini file with the contents:

```
{
  "settingUnlockOnUnequip": true,
  "settingRageSet": true,
  "settingMaxResults": 5000000,
  "settingDefaultPath": "C:\\FribbelsOptimizerSaves",
  "settingExcludeEquipped": []
}
```

And change the settingDefaultPath to your desired path

**Comment by numbahs** [comment]
Okay, it is duplicated in both my H drive and my C drive, and it still isn't taking the autosave with the same error

**Comment by fribbels** [comment]
In that case try opening the dev tools again, mess around with some settings, then CTRL + A to select the whole Console log and CTRL + C to copy it, and paste it into some sharing service like https://ctxt.io/ so I can check the log

**Comment by numbahs** [comment]
Looks like it is doing the LOAD SETTINGS and then it is trying to find the autosave, and then it is doing the LOADING SETTINGS portion.

```LOAD SETTINGS C:\Users\[USER]\Documents/FribbelsOptimizerSaves/settings.ini
H:\Installs\E7Optimizer\resources\app.asar\js\lib\grids\optimizerGrid.js:103 DEBUG getRows params Object
H:\Installs\E7Optimizer\resources\app.asar\js\lib\selectors.js:113 Set selectors Array(0) Array(0) Array(0) Array(0)
H:\Installs\E7Optimizer\resources\app.asar\js\lib\selectors.js:103 Main stat selectors Array(0) Array(0) Array(0)
H:\Installs\E7Optimizer\resources\app.asar\js\lib\selectors.js:94 Exclude Array(0)
H:\Installs\E7Optimizer\resources\app.asar\js\lib\tabs\optimizerTab.js:993 SETFORMAT 0
H:\Installs\E7Optimizer\resources\app.asar\js\lib\inputHandler.js:9 Failed to load autosave - Error: ENOENT: no such file or directory, open 'C:\Users\[USER]\Documents\FribbelsOptimizerSaves\autosave.json'
console.error @ H:\Installs\E7Optimizer\resources\app.asar\js\lib\inputHandler.js:9
H:\Installs\E7Optimizer\resources\app.asar\js\lib\settings.js:79 LOADING SETTINGS Object```

**Comment by fribbels** [comment]
Your snippet is an old version of that file - the current 1.5 version of the file looks like:

```
    Subprocess.initialize(async () => {
        await HeroData.initialize();
        ZarrocConverter.initialize();
        OptimizerTab.initialize();
        OptimizerGrid.initialize();
        ItemsTab.initialize();
        ItemsGrid.initialize();
        HeroesTab.initialize();
        HeroesGrid.initialize();

        Settings.initialize();
        Saves.loadAutoSave();
    });
```

Turns out it was an asynchronous bug - settings were being initialized first but it finished being initialized after the save was loaded in some cases. Fixing in 1.6

**Comment by numbahs** [comment]
Ah okay, I was thinking it was either async/await issues or the code, but makes sense since the code I looked at was old

**Comment by fribbels** [comment]
This specific issue has been fixed in 1.6
